File: Go/concatenation-of-consecutive-binary-numbers.go
func concatenatedBinary(n int) int {
    result := 0
    limit := 1
    mod := 1000000007
    shift := 0
    for i := 1; i <= n; i++ {
        if i >= limit {
            limit *= 2
            shift++
        }
        result = (result << shift + i) % mod
        //fmt.Printf("%b\n", result)
    }
    return result
}

File: JavaScript/concatenation-of-consecutive-binary-numbers.js
/**
 * @param {number} n
 * @return {number}
 */
var concatenatedBinary = function(n) {
    const mod = 1000000007n
    let result = 0n
    let limit = 1n
    let shift = 0n
    for (let i = 1n; i <= BigInt(n); i++) {
        if (i >= limit) {
            limit += limit
            shift++
        }
        result = ((result << shift) | i) % mod
    }
    return Number(result)
};